tools/src/launcher/Main.hs

```diff
@@ -479,33 +495,21 @@ spawnNode (path, args, mbLogPath) = do
     logNotice "Starting the node"
     -- We don't explicitly close the `logHandle` here,
     -- but this will be done when we run the `CreateProcess` built
-    -- by proc later is `system'`:
+    -- by proc later in `system'`:
     -- http://hackage.haskell.org/package/process-1.6.1.0/docs/System-Process.html#v:createProcess
-    (_, logHandle) <- liftIO $ case mbLogPath of
+    cr <- liftIO $ case mbLogPath of
         Just lp -> do
-            createDirectoryIfMissing True (directory lp)
-            (lp,) <$> openFile lp AppendMode
-        Nothing -> do
-            tempdir <- fromString <$> getTemporaryDirectory
-            -- FIXME (adinapoli): `Shell` from `turtle` was giving us no-resource-leak guarantees
-            -- via the `Managed` monad, which is something we have lost here, and we are back to manual
-            -- resource control. In this case, however, shall we really want to nuke the file? It seems
-            -- something useful to have lying around in the filesystem. We should probably close the
-            -- `Handle`, though, but if this program is short lived it won't matter anyway.
-            IO.openTempFile tempdir "cardano-node-output.log"
-    -- TODO (jmitchell): Find a safe, reliable way to print `logPath`. Cardano
-    -- fails when it prints unicode characters. In the meantime, don't print it.
-    -- See DAEF-12.
-
-    -- printf ("Redirecting node's stdout and stderr to "%fp%"\n") logPath
-    liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
-    let cr = (Process.proc (toString path) (map toString args))
-                 { Process.std_in  = Process.CreatePipe
-                 , Process.std_out = Process.UseHandle logHandle
-                 , Process.std_err = Process.UseHandle logHandle
-                 }
+            _ <- createDirectoryIfMissing True (directory lp)
+            logHandle <- openFile lp AppendMode
+            liftIO $ IO.hSetBuffering logHandle IO.LineBuffering
+            -- TODO (jmitchell): Find a safe, reliable way to print `logPath`. Cardano
+            -- fails when it prints unicode characters. In the meantime, don't print it.
+            -- See DAEF-12.
+            -- printf ("Redirecting node's stdout and stderr to "%fp%"\n") logPath
+            return $ createProc (Process.UseHandle logHandle) path args
+        Nothing -> return $ createProc Process.Inherit path args
```

Comment:
It should be `CreatePipe`, not `Inherit`, when wallet also wants to log to console.
That's actually the only case when we want to use `asyncLog`: when both executables want to log to console.
If wallet's output is redirected to a file or disabled, it should be `Inherit`.

The same is true for wallet: if node prints to console, wallet should use `CreatePipe` instead of `Inherit` (when wallet logging is enabled).